src/main/java/cz/student/simon8491/UserDao.java

```diff
@@ -0,0 +1,115 @@
+            try (var stmtCreateUser = connection.prepareStatement
+                    (CREATE_USER_QUERY, PreparedStatement.RETURN_GENERATED_KEYS)) {
+                stmtCreateUser.setString(1, user.getUserName());
+                stmtCreateUser.setString(2, user.getEmail());
+                stmtCreateUser.setString(3, hashPassword(user.getPassword()));
+                stmtCreateUser.executeUpdate();
+                try (var rsCreateUser = stmtCreateUser.getGeneratedKeys()) {
+                    if (rsCreateUser.next()) {
+                        user.setId(rsCreateUser.getInt(1));
+                        System.out.println("Created user with ID: " + user.getId());
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            System.err.println("Database connection error\n" + e.getMessage());
+        }
+    }
+
+    public String hashPassword(String password) {
+        return BCrypt.hashpw(password, BCrypt.gensalt());
+    }
+
+    public User readUser(int id) {
+        try (Connection connection = DBUtil.getConnection()) {
+            try (var stmtReadUser = connection.prepareStatement(READ_USER_QUERY)) {
+                stmtReadUser.setInt(1, id);
+                try (var rsReadUser = stmtReadUser.executeQuery()) {
+                    if (rsReadUser.next()) {
+                        User user = new User();
+                        user.setId(rsReadUser.getInt("id"));
+                        user.setUserName(rsReadUser.getString("username"));
+                        user.setEmail(rsReadUser.getString("email"));
+                        user.setPassword(rsReadUser.getString("password"));
+                        return user;
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            System.err.println("Database connection error\n" + e.getMessage());
+        }
+        return null;
+    }
+
+    public void updateUser(User user) {
+        try (Connection connection = DBUtil.getConnection()) {
+            try (var stmtUpdateUser = connection.prepareStatement(UPDATE_USER_QUERY)) {
+                stmtUpdateUser.setString(1, user.getUserName());
+                stmtUpdateUser.setString(2, user.getEmail());
+                stmtUpdateUser.setString(3, this.hashPassword(user.getPassword()));
+                stmtUpdateUser.setInt(4, user.getId());
+                stmtUpdateUser.executeUpdate();
+            }
+        } catch (SQLException e) {
+            System.err.println("Database connection error\n" + e.getMessage());
+        }
+    }
+
+    public void deleteUser(int id) {
+        try (Connection connection = DBUtil.getConnection()) {
+            try (var stmtDeleteUser = connection.prepareStatement(DELETE_USER_QUERY)) {
+                stmtDeleteUser.setInt(1, id);
+                stmtDeleteUser.executeUpdate();
+                System.out.println("Deleted user with ID: " + id);
```

Comment:
IMPROVE: zalogovat informaci o tom jestli byl opravdu uzivatel smazan.

```suggestion
            int affectedRows = preStmt.executeUpdate();
            if (affectedRows > 0) {
                System.out.println("User with id %s deleted successfully.".formatted(userId));
            } else {
                System.out.println("User delete with id %s failed. No rows affected.".formatted(userId));
            }
```